### PR TITLE
add support for LZW compression

### LINKED ISSUE #8: Add support for LZW compression
LZW compression seems very commonly used with TIFF so it would be nice to support going forward (ref #6).

### PR DESCRIPTION
**Pull request by chrstphrbrns**
Fixes #8 

Validated using images from a variety of packages, using a variety of pixel formats, and comparing with uncompressed versions of the same image

### CODE REVIEW

**Comment by tlnagy** [comment]
Wow, thanks! LZW support has been requested for awhile. 

I'll review this carefully later this week, but do you mind adding tests? I believe `bali.tif` from my tlnagy/exampletiffs repo is LZW compressed. 

Also is this your own original implementation or was this adapted from somewhere? We just need to make sure the licenses are compatible if the latter.

**Comment by chrstphrbrns** [comment]
This implementation is not adapted or ported from any other code

File: src/ifds.jl [hunk 1]
```diff
@@ -203,6 +203,15 @@ function Base.iterate(file::TiffFile, state::Tuple{Union{IFD{O}, Nothing}, Int})
     return (curr_ifd, (next_ifd, next_ifd_offset))
 end
 
+struct TiffFileStrip{T}
+    tf::TiffFile
+    ifd::IFD
+    bytes::Int
+end
```
Comment (tlnagy) [inline src/ifds.jl:210]: Do you mind adding docs to `TiffFileStrip`?

**Comment by tlnagy** [comment]
Do you mind bumping the version to `v0.7.0`?

**Review by tlnagy — CHANGES_REQUESTED** [review]

File: src/ifds.jl [hunk 2]
```diff
@@ -203,6 +203,27 @@ function Base.iterate(file::TiffFile, state::Tuple{Union{IFD{O}, Nothing}, Int})
...
+"""
+    $(TYPEDEF)
+
+A strip is a contiguous block of separately-encoded image data. A TIFF
+file will typically have multiple strips, each representing multiple rows of
+pixels in the image
+
+$(FIELDS)
+"""
+struct TiffFileStrip{T}
```
[the remark this answers is not in the document]
Comment (chrstphrbrns) [reply src/ifds.jl:215]: `T` is the sample type, which is neither `O` nor `S`. It was added to address type instability in `reverse_prediction!`

[the remark this answers is not in the document]
Comment (tlnagy) [reply src/ifds.jl]: Ah that makes sense. The 3 parameter type should do the trick, without it @code_warntype was complaining about `TiffFileStrip`.

**Review by tlnagy — APPROVED** [review]

**Comment by tlnagy** [comment]
Okay, I think this is good to go. I'll tag v0.6.8 first to release #121 and then we'll tag and release this PR as v0.7.0

**Comment by tlnagy** [comment]
Thanks @chrstphrbrns for pushing this all the way through! I think people will be very excited for this.

**Pull request merged**